File: 백준/Gold/5430. AC/AC.js
let [T, ...input] = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n').map(line => line.trim());
T = Number(T);
const ans = [];
outerLoop:
for(let tc = 0; tc < T; tc++) {
  const fuc = input[tc * 3].split('');
  const N = Number(input[tc * 3 + 1]);
  const arr = input[tc * 3 + 2] !== '[]' ? input[tc * 3 + 2].substring(1, input[tc * 3 + 2].length - 1).split(',').map(Number) : [];
  let flag = false;
  let start = 0;
  let end = N - 1;
  for(let i = 0; i < fuc.length; i++) {
    if(fuc[i] === 'R') {
      flag = !flag;
    } else {
      if(start > end) {
        ans.push('error');
        continue outerLoop;
      }
      if(!flag) start++;
      else end--;
    }
  }
  let str = '[';
  let tmp = [];
  if(!flag) {
    for(let i = start; i <= end; i++) {
      tmp.push(arr[i]);
    }
  } else {
    for(let i = end; i >= start; i--) {
      tmp.push(arr[i]);
    }
  }
  str += tmp.join(',') + ']';
  ans.push(str);
}
console.log(ans.join('\n'));

File: 백준/Gold/5430. AC/AC.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int T = Integer.parseInt(br.readLine());
        outerLoop:
        for(int tc = 0; tc < T; tc++) {
            char[] fuc = br.readLine().toCharArray();
            int N = Integer.parseInt(br.readLine());
            String tmp = br.readLine();
            String[] tmp_arr = tmp.substring(1, tmp.length() - 1).split(",");
            Deque<Integer> que = new ArrayDeque<>();
            for(int i = 0; i < N; i++) {
                que.add(Integer.parseInt(tmp_arr[i]));
            }
            boolean flag = false;
            for(char now : fuc) {
                if(now == 'R') flag = !flag;
                else {
                    if (que.isEmpty()) {
                        sb.append("error\n");
                        continue outerLoop;
                    }
                    if(!flag) que.removeFirst();
                    else que.removeLast();
                }
            }
            sb.append("[");
            if (!que.isEmpty()) {
                if (flag) {
                    while (que.size() > 1) sb.append(que.pollLast()).append(",");
                    sb.append(que.pollLast());
                } else {
                    while (que.size() > 1) sb.append(que.pollFirst()).append(",");
                    sb.append(que.pollFirst());
                }
            }
            sb.append("]\n");
        }
        System.out.println(sb.toString().trim());
    }
}
